### PR TITLE
Give toyotas a little boost when starting

### PR DESCRIPTION
**Pull request by haraschax**

### CODE REVIEW

File: selfdrive/car/toyota/carcontroller.py [hunk 1]
```diff
@@ -42,12 +42,17 @@ def update(self, CC, CS):
...
-      pedal_offset = interp(CS.out.vEgo, [0.0, 2.3, MIN_ACC_SPEED + PEDAL_TRANSITION], [-.4, 0.0, 0.2])
+      pedal_offset = interp(CS.out.vEgo, [0.0, CREEP_SPEED, MIN_ACC_SPEED + PEDAL_TRANSITION], [-.4, 0.0, 0.2])
       pedal_command = PEDAL_SCALE * (actuators.accel + pedal_offset)
       interceptor_gas_cmd = clip(pedal_command, 0., MAX_INTERCEPTOR_GAS)
+      boost = 0.0
     else:
       interceptor_gas_cmd = 0.
-    pcm_accel_cmd = clip(actuators.accel, CarControllerParams.ACCEL_MIN, CarControllerParams.ACCEL_MAX)
+      # Toyotas don't respond to small accel requests when stationary
+      start_boost = interp(CS.out.vEgo, [0.0, CREEP_SPEED, 2*CREEP_SPEED], [0.6, 0.6, 0.0])
```
Comment (sshane) [inline selfdrive/car/toyota/carcontroller.py:52]:
```suggestion
      start_boost = interp(CS.out.vEgo, [0.0, CREEP_SPEED, 2 * CREEP_SPEED], [0.6, 0.6, 0.0])
```

File: selfdrive/car/toyota/carcontroller.py [hunk 2]
```diff
@@ -42,12 +42,17 @@ def update(self, CC, CS):
       else:
         PEDAL_SCALE = interp(CS.out.vEgo, [0.0, MIN_ACC_SPEED, MIN_ACC_SPEED + PEDAL_TRANSITION], [0.4, 0.5, 0.0])
       # offset for creep and windbrake
-      pedal_offset = interp(CS.out.vEgo, [0.0, 2.3, MIN_ACC_SPEED + PEDAL_TRANSITION], [-.4, 0.0, 0.2])
+      pedal_offset = interp(CS.out.vEgo, [0.0, CREEP_SPEED, MIN_ACC_SPEED + PEDAL_TRANSITION], [-.4, 0.0, 0.2])
```
Comment (sshane) [inline selfdrive/car/toyota/carcontroller.py:45]:
```suggestion
      pedal_offset = interp(CS.out.vEgo, [0.0, CREEP_SPEED, MIN_ACC_SPEED + PEDAL_TRANSITION], [-0.4, 0.0, 0.2])
```

**Review by sshane — COMMENTED** [review]
just minor formatting

File: selfdrive/car/toyota/carcontroller.py [hunk 3]
```diff
@@ -42,12 +42,17 @@ def update(self, CC, CS):
...
       interceptor_gas_cmd = clip(pedal_command, 0., MAX_INTERCEPTOR_GAS)
+      boost = 0.0
     else:
       interceptor_gas_cmd = 0.
-    pcm_accel_cmd = clip(actuators.accel, CarControllerParams.ACCEL_MIN, CarControllerParams.ACCEL_MAX)
+      # Toyotas don't respond to small accel requests when stationary
+      start_boost = interp(CS.out.vEgo, [0.0, CREEP_SPEED, 2 * CREEP_SPEED], [0.6, 0.6, 0.0])
+      is_accelerating = interp(actuators.accel, [0.0, 0.2], [0.0, 1.0])
+      boost = start_boost * is_accelerating
+    pcm_accel_cmd = clip(actuators.accel, CarControllerParams.ACCEL_MIN, CarControllerParams.ACCEL_MAX) + boost
```
Comment (cydia2020) [inline selfdrive/car/toyota/carcontroller.py:55]:
```python
pcm_accel_cmd = clip(actuators.accel + boost, CarControllerParams.ACCEL_MIN, CarControllerParams.ACCEL_MAX)
```

Comment (HaraldSchafer) [reply selfdrive/car/toyota/carcontroller.py]: ty

**Comment by cydia2020** [comment]
Tested this on my Prius, the boost is a bit too strong, which results in accelerator/brake oscillations

**Comment by HaraldSchafer** [comment]
Hmmm, I like it on the TSS2 Corolla, maybe it's car dependent

**Comment by sshane** [comment]
Yeah it's possible `EV_HYBRID_CAR` needs to be down-tuned

**Comment by nelsonjchen** [comment]
Could this be rebased/merged/etc. on a recent master that has this commit when time permits? 

https://github.com/commaai/openpilot/commit/29afd53d88f5c98efdac8727f3df1e4679790cde

I just switched back to master-ci for the moment but I would still like to be contributing data for this PR.

**Comment by sshane** [comment]
Unfortunately when you engage downward on a hill, you can really feel the ramp down of acceleration as you leave the interp range, I'll post a route. Also had a case where the car overshot badly when the lead creeped up

**Comment by HaraldSchafer** [comment]
Closing, going to make an explicit starting state

**Pull request closed without merging**